### PR TITLE
Airbyte CDK: request options allowed to be an array

### LINKED ISSUE #36358: Airbyte CDK: request option parameters are allowed to be an array
## The problem

In some cases it is required the request options to be an array of objects:
```
request_parameters:
    expand: "description,lead"
    status: ["live", "archived", "deleted"]
```
which should be expanded to the following url structure:
```
https://<base-url>?
  status=live&
  status=archived&
  status=deleted&
  exapnd=...
```

## Solution

Correct the validation rules of HTTPRequester to permit request options to accept values of the list type

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
Request options allowed to be an array:
```
request_parameters:
  status: ["live", "archived", "deleted"]
```

### CODE REVIEW

File: airbyte-cdk/python/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 0.73.1
```
Comment (girarda) [inline airbyte-cdk/python/CHANGELOG.md:3]: no need to update the changelog. this is done automatically by the publish github action

File: airbyte-cdk/python/.bumpversion.cfg [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 0.73.0
+current_version = 0.73.1
```
Comment (girarda) [inline airbyte-cdk/python/.bumpversion.cfg:2]: no need to bump the version. this is done automatically by the publish github action

File: airbyte-cdk/python/Dockerfile [hunk 3]
```diff
@@ -10,7 +10,7 @@ RUN apk --no-cache upgrade \
     && apk --no-cache add tzdata build-base
 
 # install airbyte-cdk
-RUN pip install --prefix=/install airbyte-cdk==0.73.0
+RUN pip install --prefix=/install airbyte-cdk==0.73.1
```
Comment (girarda) [inline airbyte-cdk/python/Dockerfile:13]: no need to bump the version. this is done automatically by the publish github action

File: airbyte-cdk/python/setup.py [hunk 4]
```diff
@@ -36,7 +36,7 @@
     name="airbyte-cdk",
     # The version of the airbyte-cdk package is used at runtime to validate manifests. That validation must be
     # updated if our semver format changes such as using release candidate versions.
-    version="0.73.0",
+    version="0.73.1",
```
Comment (girarda) [inline airbyte-cdk/python/setup.py:39]: no need to bump the version. this is done automatically by the publish github action

File: airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py [hunk 5]
```diff
@@ -386,67 +386,67 @@ def test_send_request_params(provider_params, param_params, authenticator_params
...
-            "k",
+            "k=a%2Cb",
             id="test-request-parameter-from-config-comma-separated-strings",
         ),
         pytest.param(
             {'["a", "b"]': '{{ config["k"] }}'},
             {"k": [1, 2]},
-            '["a", "b"]',
-            id="test-key-with-list-is-not-interpolated",
+            "%5B%22a%22%2C+%22b%22%5D=1&%5B%22a%22%2C+%22b%22%5D=2",
```
Comment (girarda) [inline airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py:422]: I'm a little confused by this test case. Why is this one being rendered as an array whereas the other cases are turning into multiple query params?

**Review by girarda — CHANGES_REQUESTED** [review]
the change itself looks good, but there's no need to update the CDK version and the changelog. that'll be done automatically as part of the CDK publish github action

File: airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py (hunk 5 above)
Comment (roman-yermilov-gl) [reply airbyte-cdk/python/unit_tests/sources/declarative/requesters/test_http_requester.py:422]: it is not an array. I think the special characters of the key itslef are rendered to this form `%5B%22a%22%2C+%22b%22%5D`

File: airbyte-cdk/python/CHANGELOG.md (hunk 1 above)
Comment (roman-yermilov-gl) [reply airbyte-cdk/python/CHANGELOG.md]: Fixed

File: airbyte-cdk/python/.bumpversion.cfg (hunk 2 above)
Comment (roman-yermilov-gl) [reply airbyte-cdk/python/.bumpversion.cfg]: Fixed

File: airbyte-cdk/python/Dockerfile (hunk 3 above)
Comment (roman-yermilov-gl) [reply airbyte-cdk/python/Dockerfile]: Fixed

File: airbyte-cdk/python/setup.py (hunk 4 above)
Comment (roman-yermilov-gl) [reply airbyte-cdk/python/setup.py]: Fixed

**Review by girarda — APPROVED** [review]
:shipit:

**Pull request merged**